app/scripts/metamask-controller.js

```diff
@@ -484,13 +480,26 @@ export default class MetamaskController extends EventEmitter {
     });
 
     this.selectedNetworkController = new SelectedNetworkController({
+      state: initState.SelectedNetworkController,
       messenger: this.controllerMessenger.getRestricted({
         name: 'SelectedNetworkController',
         allowedActions: ['NetworkController:getNetworkClientById'],
         allowedEvents: ['NetworkController:stateChange'],
       }),
     });
 
+    if (
+      !this.selectedNetworkController.getNetworkClientIdForDomain('metamask')
```

Comment:
Should we pull in and use the `ORIGIN_METAMASK` constant here, using the same variable as the SelectedNetworkController itself. Currently its not exported from the `SelectedNetworkController` so won't block on this but maybe we should export it?